CMakeLists.txt

```diff
@@ -277,53 +237,19 @@ if (ARMA_EXTRA_DEBUG)
   add_definitions(-DARMA_EXTRA_DEBUG)
 endif()
 
-# Now, find the libraries we need to compile against.  Several variables can be
-# set to manually specify the directory in which each of these libraries
-# resides.
-#   ARMADILLO_LIBRARY - location of libarmadillo.so / armadillo.lib
-#   ARMADILLO_INCLUDE_DIR - directory containing <armadillo>
-#   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
-#   CEREAL_INCLUDE_DIR - include directory for cereal
-#   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
-#   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
-#   MATHJAX_ROOT - root of MathJax installation
-
-# Download and compile OpenBLAS if we are cross compiling mlpack for a specific
-# architecture. The function takes the version of OpenBLAS as variable.
 if (CMAKE_CROSSCOMPILING)
-  search_openblas(0.3.28)
-endif()
-
-if (NOT DOWNLOAD_DEPENDENCIES)
-  find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
+  fetch_mlpack_and_crosscompile()
```

Comment:
I think that you changed this function and removed it, so should this be removed here, and we can just call something like this?

```cmake
if (CMAKE_CROSSCOMPILING OR DOWNLOAD_DEPENDENCIES)
  fetch_mlpack(ON)
else ()
  find_mlpack()
endif ()
```